src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -285,9 +316,33 @@ public synchronized IndexShard createShard(int sShardId, boolean primary) {
         Injector shardInjector = null;
         try {
 
+            
+
             ShardPath path = ShardPath.loadShardPath(logger, nodeEnv, shardId, indexSettings);
             if (path == null) {
-                path = ShardPath.selectNewPathForShard(nodeEnv, shardId, indexSettings);
+
+                long totFreeSpace = 0;
```

Comment:
any change we can move this heuristic into a utils class or into `ShardPath` as a static method?